packages/core/src/app/shipping/NewConsignment.tsx

```diff
@@ -45,16 +54,63 @@ const NewConsignment = ({
         };
     }, [consignmentRequest]);
 
+
+    const toggleAllocateItemsModal = () => {
+        setIsOpenAllocateItemsModal(!isOpenAllocateItemsModal);
+    }
+
+    const handleAllocateItems = async (consignmentLineItems: ConsignmentLineItem[]) => {
+        if (!selectedAddress) {
+            return;
+        }
+
+        try {
+            await assignItem({
+                address: selectedAddress,
+                lineItems: consignmentLineItems,
+            });
+            setIsAddShippingDestination(false);
+        } catch (error) {
+            if (error instanceof AssignItemFailedError) {
+                onUnhandledError(error);
+            }
+        } finally {
+            toggleAllocateItemsModal();
+        }
+    }
+
     return (
         <div className='consignment-container'>
-            <h3 className='consignment-header'>Shipping destination {consignmentNumber}</h3>
+            <h3 className='consignment-header'>Destination #{consignmentNumber}</h3>
```

Comment:
Oooops, in order to support multi-lang, we need to put new strings into `en.json`, and use it in this way `<TranslatedString id="address.add_address_heading" />`. 
https://github.com/bigcommerce/checkout-js/blob/c21be1be1b29d589e6473490112601ec469d8887/packages/locale/src/translations/en.json#L2

This also applies to places like `No items allocated` and `Allocate items` below.

Reply:
I've created this ticket to localise all the strings for this project https://bigcommercecloud.atlassian.net/browse/CHECKOUT-8780

Reply:
This is ok for now, but ideally we should localise the strings as we add them.

Reply:
Thought we could do the locale ticket right after this PR in order to keep the gap small.